lib/ASRouter.jsm

```diff
@@ -1767,7 +1767,62 @@ class _ASRouter {
           csp: null,
         });
         break;
+      case ra.ENABLE_FIREFOX_MONITOR:
+        const {
+          utm_term,
+          utm_source,
+          utm_campaign,
+          entrypoint,
+          form_type,
+        } = action.data;
+        const { flowBeginTime, flowId } = await this.fetchFlowParams({
+          entrypoint,
+          utm_term,
+          form_type,
+        });
+        const monitorURL = `https://monitor.firefox.com/oauth/init?utm_source=${utm_source}&utm_campaign=${utm_campaign}&form_type=${form_type}&entrypoint=${entrypoint}&flowBeginTime=${flowBeginTime}&flowId=${flowId}`;
```

Comment:
I'm curious what you think about passing the whole URL with templates for `flowBeginTime` and `flowId` (which have to come from Firefox) v.s. passing variables:

With templated URL: 
```json
{
  "button_action": "ENABLE_FIREFOX_MONITOR",
  "button_action_args": "https://monitor.firefox.com/oauth/init?utm_source=snippets&utm_campaign=foo&${flowBeginTime}&flowId=${flowId}"
}
```

With variables: 
```json
{
  "button_action": "ENABLE_FIREFOX_MONITOR",
  "button_action_args": {
    "utm_source": "snippets",
    "utm_campaign": "foo"
  }
}
```

I think passing the whole URL could give you more flexibility in changing different parts of the URL remotely, but variables might be less error-prone.

Reply:
I prefer URL for the flexibility. Either way works fine for me. Variables or URL will be handled in the backend and will not be directly editable by Snippet Editors so errors will be limited. Thanks for reaching out! :cake: